src/dramatiq_abort/middleware.py

```diff
@@ -157,17 +180,19 @@ def _watcher(self) -> None:
                 )
 
     @staticmethod
-    def id_to_key(message_id: str) -> bytes:
-        return ("abort:" + message_id).encode()
+    def id_to_key(message_id: str, mode: AbortMode = AbortMode.ABORT) -> bytes:
+        return (mode.value + ":" + message_id).encode()
 
     @staticmethod
     def key_to_id(key: bytes) -> str:
-        return key.decode()[6:]
+        return key.split(b":")[-1].decode()
```

Comment:
Just for safety, let's use `split(b":", 1)`. This way an id with `:` will still work.